base/src/main/java/com/smartdevicelink/proxy/rpc/SystemCapability.java

```diff
@@ -115,4 +117,23 @@ public void setCapabilityForType(SystemCapabilityType type, RPCStruct capability
 	        return;
         }
     }
+
+    /**
+     * set the displayCapabilities portion of SystemCapability.
+     * {@code 1<= displayCapabilities.size() <= 100}
+     *
+     * @param displayCapabilities Contains a list of capabilities of display.
+     */
+    public void setDisplayCapabilities(List<DisplayCapability> displayCapabilities) {
```

Comment:
These two methods should not be added because `getCapabilityForType()`, `setCapabilityForType()` should work for all capabilties incldunbg DISPLAYS

Reply:
Hi @bilal-alsharifi thanks for the recommendation, but the method  `setCapabilityForType()` receives a RPCStruct. `setDisplayCapabilities()` receives a `List<DisplayCapability>`. 
What is your recommendation?

Reply:
You are right. I think the best thing we can do is to change the `getCapabilityForType()`,  and `setCapabilityForType()` to return/take  java `Object` instead of `RPCStruct`. I explained more about that in other comments on the same `SystemCapability.java` file. What do you think?

Reply:
@bilal-alsharifi because the method is public this change is difficult to make. I'm not sure loosing type parameter type from an `Object` subclass to `Object` causes breaking changes but again... they are public methods.

Reply:
You are right they are public.  But because `Object` is more generic than `RPCStruct` the change shouldn't break existing code because most developers will cast to the capability type anyway like the following :
```
RemoteControlCapabilities remoteControlCapabilities = (RemoteControlCapabilities) systemCapability.getCapabilityForType(SystemCapabilityType.REMOTE_CONTROL);
```
That will keep working whether the return type is `RPCStruct` or `Object`. I know cases like the following will break. But I think the most developer will not even do that:
```
RPCStruct rpcStruct = systemCapability.getCapabilityForType(SystemCapabilityType.REMOTE_CONTROL);
```
It should only be a compile-time error that devs can easily fix by casting to the right type.

I am looking at the internal uses for that method in the library just to have in an idea for developer will often use the method.  they will all not break.
https://github.com/smartdevicelink/sdl_java_suite/blob/014401c0dc52f185bd73431b6e386fd59e1a3c07/base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java#L111
and
https://github.com/smartdevicelink/sdl_java_suite/blob/014401c0dc52f185bd73431b6e386fd59e1a3c07/android/sdl_android/src/androidTest/java/com/smartdevicelink/test/proxy/SystemCapabilityManagerTests.java#L138

I do understand your concerns though. But also if we don't handle the `DISPLAYS` capability in that method, it will return null even if the capability exists which can lead to more serious run time error. also having a generic method for all types except the `DISPLAYS` sounds is a little confusing for the developers.

I know that changing that public method is not a great way but I think its pros outweigh the cons.  
The other option that we can do is deprecating the current methods and add new ones that take/return `Object`